### PR TITLE
Pass infra base environment settings from Agent to Vertex job

### PR DESCRIPTION
**Pull request by salman1993**
The Vertex agent does not currently set the Prefect API URL or API Key when the Vertex job is launched. This means the launched Vertex job cannot report the state to the Prefect API. This is probably an oversight because:
1. This used to be set in Prefect 1.x VertexAgent: https://github.com/slate-data/prefect/blob/a6c930403642c35d34d2ccb4b508cc85ce0566aa/src/prefect/agent/vertex/agent.py#L215 
2. This is currently set in the `prefect-aws/ecs.py -> ECSTask`:  https://github.com/PrefectHQ/prefect-aws/blob/6fe4c564dc865e3500e663a99b346fb515222d07/prefect_aws/ecs.py#L1344 

Changes:
- Add base env variables to pass settings from agent to Vertex job
- Fix test_preview by adding in the `env` string

Closes

### Example

Tested this out and it adds in the 2 profile env vars as expected - `PREFECT_API_URL` & `PREFECT_API_KEY`.

### Checklist

- [ ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-gcp/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [ ] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-gcp/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_gcp/aiplatform.py [hunk 1]
```diff
@@ -204,7 +204,13 @@ def _build_job_spec(self) -> "CustomJobSpec":
         Builds a job spec by gathering details.
         """
         # gather worker pool spec
-        env_list = [{"name": name, "value": value} for name, value in self.env.items()]
+        env_list = [
+            {"name": name, "value": value}
+            for name, value in {
+                **self._base_environment(),
```
Comment (madkinsz) [inline prefect_gcp/aiplatform.py:210]: 👍 this is definitely essential for any infrastructure block

File: tests/test_aiplatform.py [hunk 2]
```diff
@@ -23,30 +23,61 @@ def vertex_ai_custom_training_job(self, gcp_credentials):
...
                     replica_count: 1
-                }
-                scheduling {
-                }
+                }}
+                scheduling {{
+                }}
                 service_account: "my_service_account_email"
-            }
+            }}
```
Comment (ahuang11) [inline tests/test_aiplatform.py:52]: I was wondering why it's now doubly nested in squiggly brackets. Is this expected?

**Review by ahuang11 — APPROVED** [review]
Thanks for helping us patch this up! Just a minor question about why it's doubly nested now in preview and a request to fix the Windows test:

```

    def test_preview(self, vertex_ai_custom_training_job: VertexAICustomTrainingJob):
        actual_lines = vertex_ai_custom_training_job.preview().splitlines()
    
        base_env = VertexAICustomTrainingJob._base_environment().copy()
        env_lines = [
            f'env {{\n        name: "{name}"\n        value: "{value}"\n      }}'
            for name, value in base_env.items()
        ]
        env_str = "\n".join(env_lines)
    
        expected_lines = f"""
            display_name: "container
            job_spec {{
                worker_pool_specs {{
                    container_spec {{
                        image_uri: "us-docker.pkg.dev/cloudrun/container/job:latest"
                        command: "echo"
                        command: "hello!!"
                        {env_str}
                    }}
                    machine_spec {{
                        machine_type: "n1-standard-4"
                    }}
                    replica_count: 1
                }}
                scheduling {{
                }}
                service_account: "my_service_account_email"
            }}
        """.strip().splitlines()
    
        for actual_line, expected_line in zip(actual_lines, expected_lines):
            if '"container' in actual_line:
                actual_line = actual_line.split("-")[0]  # remove the unique hex
>           assert actual_line.strip() == expected_line.strip()  # disregard whitespace
E           assert 'value: "sqlite+aiosqlite:///C:\\\\Users\\\\RUNNER~1\\\\AppData\\\\Local\\\\Temp\\\\tmp8craaxer\\\\orion.db"' == 'value: "sqlite+aiosqlite:///C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmp8craaxer\\orion.db"'
E             - value: "sqlite+aiosqlite:///C:\Users\RUNNER~1\AppData\Local\Temp\tmp8craaxer\orion.db"
E             + value: "sqlite+aiosqlite:///C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmp8craaxer\\orion.db"
```
Seems like mismatch of `\`

File: tests/test_aiplatform.py (hunk 2 above)
Comment (salman1993) [reply tests/test_aiplatform.py:52]: We are escaping the `{` here. This is required when using a f-string with multi-line strings

Comment (ahuang11) [reply tests/test_aiplatform.py:52]: Ah that makes sense. Thanks!

Comment (salman1993) [reply tests/test_aiplatform.py:52]: Do you know if there is an easy way to make paths work for the Linux & Windows test?

Comment (ahuang11) [reply tests/test_aiplatform.py:52]: Perhaps wrapping this utility to compare actual_line vs expected_line:
https://github.com/PrefectHQ/prefect/blob/main/src/prefect/utilities/filesystem.py#L113-L119

Comment (salman1993) [reply tests/test_aiplatform.py]: I cleaned up the tests by patching the returned value of `_base_environment()`

**Review by ahuang11 — APPROVED** [review]

**Comment by ahuang11** [comment]
Thanks again for the fix!

**Pull request merged**